docs/spec_draft.md

```diff
@@ -326,6 +327,84 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.broadcast_in_dim
+
+### Semantics
+
+Expands the dimensions and/or rank of an input tensor by duplicating the data
+in the `operand` tensor and produces a `result` tensor. Formally, 
+`result[i0, i1, ..., iR-1]` $=$ `operand[j0, j1, ..., jR-1]` such that
+`jk` $=$ `dim(operand, k) == 1 ? 0 : i[boradcast_dimensions[k]]`, for all
+dimensions `k` in `operand`.
+  
+### Operands
+
+| Name                   | Type                                  |
+|------------------------|---------------------------------------|
+| `operand`              | tensor of any supported element types |
+| `broadcast_dimensions` | 1-dimensional array of type `si64`    |
+
+### Results
+
+| Name     | Type                                  |
+|----------|---------------------------------------|
+| `result` | tensor of any supported element types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the element type.
+  * (C2) dim(`broadcast_dimensions`, 0) $=$ rank(`operand`).
```

Comment:
`size(broadcast_dimensions) = rank(operand)`? Seems to be a bit shorter than `dim(broadcast_dimensions, 0)`, and is probably frequent enough to warrant having a separate function? E.g. `concatenate`'s (C2) could be rewritten using `size` as well.

Reply:
Agreed about the use of size on 1-D tensors. Added an action task to define these terms  in  https://github.com/openxla/stablehlo/issues/289